src/Library/demos/Preferences Window/main.js

```diff
@@ -0,0 +1,38 @@
+import Adw from "gi://Adw";
+import Gtk from "gi://Gtk";
+
+const pref_window = workbench.builder.get_object("pref_window");
+const dm_switch = workbench.builder.get_object("dm_switch");
+const subpage = workbench.builder.get_object("subpage");
+const subpage_row = workbench.builder.get_object("subpage_row");
+const subpage_button = workbench.builder.get_object("subpage_button");
+const toast_button = workbench.builder.get_object("toast_button");
+const style_manager = Adw.StyleManager.get_default();
+
+dm_switch.active = style_manager.dark;
+
+dm_switch.connect("state-set", (widget, state) => {
```

Comment:
this is not the right thing to connect to

please use `notify:active` signal and `switch.active` property